### ISSUE TITLE
ClientSession with default params caches DNS entries indefinitely

### ISSUE THREAD

**Issue by zackelan** [aio-libs/aiohttp#1914]
## Long story short

ClientSession and TCPConnector, with its default parameters, caches DNS entries, but allegedly only for 10 seconds (https://github.com/aio-libs/aiohttp/blob/master/aiohttp/connector.py#L563). However, it seems to be caching the responses indefinitely by default.

## Expected behaviour

When the result of a DNS lookup changes, the cached entry should eventually expire and a new lookup should be performed.

## Actual behaviour

The old DNS entry seems to be getting cached forever.

## Steps to reproduce

This originally happened when a remote server changed IPs, but I was able to reproduce it locally using Docker, an Nginx container to act as the server, and this simple test script:

test.py:

```
import asyncio
import logging
import sys

import aiohttp

def main():
    logging.basicConfig(level=logging.INFO, format="%(asctime)s %(message)s")

    url = sys.argv[1]
    logging.info(url)

    async def test():
        session = aiohttp.ClientSession()

        # If you construct the session this way, it will work even through DNS changes
        #connector = aiohttp.TCPConnector(use_dns_cache=False)
        #session = aiohttp.ClientSession(connector=connector)

        while True:
            try:
                async with session.get(url, timeout=1) as response:
                    logging.info(response.status)
            except Exception:
                logging.exception("error")
            finally:
                await asyncio.sleep(1)

    loop = asyncio.get_event_loop()
    loop.run_until_complete(test())

if __name__ == '__main__':
    main()
```

Dockerfile:

```
FROM python:3.6.1-slim
RUN pip install aiohttp==2.0.7
COPY test.py /srv/test.py
CMD ["python3", "/srv/test.py", "http://test-server"]
```

Commands:

```
docker network create --subnet 192.168.42.0/24 test-net
docker run -d --net=test-net --ip 192.168.42.100 --name test-server nginx:latest

docker build -t test-client .
docker run -d --net=test-net --name test-client test-client

# At this point, test-client is behaving as expected - each request is coming back with a 200 status:
docker logs test-client

# Now, change the IP of test-server from xxx.100 to xxx.200:
docker rm -f test-server
docker run -d --net=test-net --ip 192.168.42.200 --name test-server nginx:latest

# In the logs of test-client now, there are TimeoutErrors and "Connect call failed ('192.168.42.100', 80)"
# This goes on well past the 10 second cache TTL that TCPConnector uses as its default.
docker logs test-client

# You can ping test-server from within the test-client container, to confirm DNS is now returning the .200 address:
docker exec -it test-client ping -c 1 test-server
```

## Your environment

Originally discovered with Python 3.6.0 and aiohttp 2.0.7. Repro steps above test against 3.6.1.

I'm using Docker 17.03.1-ce on Linux 4.9.25, but don't believe that should affect the repro steps.

**Comment by asvetlov** [comment]
Not sure if I follow.
Would you point on problematic lines in aiohttp sources?
Or maybe even better -- create a Pull Request with fix?

**Comment by zackelan** [comment]
I was looking at the wrong branch...apparently this is already fixed in the master branch by actually expiring entries:
https://github.com/aio-libs/aiohttp/blob/master/aiohttp/connector.py#L663

However, both the 1.3 series (which most of our services currently use) or 2.0.7 (which we're working on upgrading to) make no attempt to ever expire cached entries:
https://github.com/aio-libs/aiohttp/blob/1.3/aiohttp/connector.py#L641
https://github.com/aio-libs/aiohttp/blob/2.0/aiohttp/connector.py#L610

This is....pretty horrifically broken behavior to have as a default. I'd love to see that change included in 2.0.8 sooner rather than later. In the meantime setting use_dns_cache to False seems to be a working workaround.

**Comment by asvetlov** [comment]
Sorry, we don't  support 1.3 anymore
Please wait for upcoming 2.0 release

**Comment by asvetlov** [comment]
Fixed by aiohttp 2.1.0
